src/main/java/br/com/dandrade/viagens/models/Flight.java

```diff
@@ -0,0 +1,56 @@
+package br.com.dandrade.viagens.models;
+
+import javax.persistence.*;
+import javax.validation.constraints.NotNull;
+import javax.validation.constraints.Positive;
+import javax.validation.constraints.Size;
+import java.util.LinkedList;
+import java.util.List;
+
+@Entity
+public class Flight {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    @ManyToOne
+    private Company company;
+
+    @NotNull
+    @Positive
+    private Integer numberOfSeats;
+
+    @Size(min = 1)
+    @ElementCollection
+    private List<Stretch> stretchs = new LinkedList<>();
```

Comment:
Se a gente for ser detalhista, não é bem uma lista que queremos aqui. Já que não permitimos trechos repetidos para um mesmo vôo. Por esse lado um melhor seria um Set. Aqui eu acho um ponto importante. Precisamos manter a semântica do nosso domínio, para maximizar a chance dele funcionar de maneira correta em qualquer tipo de cenário. 

Tenta fazer essa alteração? Claro que tem o desafio da ordem também.

Reply:
E eu até fico feliz que tenha alguma coisa para a gente debater huahauha.

Reply:
> Se a gente for ser detalhista, não é bem uma lista que queremos aqui. Já que não permitimos trechos repetidos para um mesmo vôo. Por esse lado um melhor seria um Set. Aqui eu acho um ponto importante. Precisamos manter a semântica do nosso domínio, para maximizar a chance dele funcionar de maneira correta em qualquer tipo de cenário.
> 
> Tenta fazer essa alteração? Claro que tem o desafio da ordem também.

Sim, vou ajustar. Acabei não considerando essa situação  de trechos iguais.



> E eu até fico feliz que tenha alguma coisa para a gente debater huahauha.

rsrsr. Nessa tarefa eu tinha certeza que ia haver pontos de ajustes... pensei muito em como implementar e no final não gostei muito, penso que poderia ficar melhor, mas foi o jeito mais simples que imaginei.